File: lua/1047.lua
local h1 = io.read("*n")
local m1 = io.read("*n")
local h2 = io.read("*n")
local m2 = io.read("*n")

local hours = h2 - h1

if hours < 0 then
    hours = 24 + hours
end

local mins = m2 - m1

if mins < 0 then
    hours = hours - 1
    mins = 60 + mins
end

if hours < 0 then
    hours = 24 + hours
end

if hours == 0 and mins == 0 then
    hours = 24
end

print(string.format("O JOGO DUROU %d HORA(S) E %d MINUTO(S)", hours, mins))

File: clojure/1047.clj
(defn calculate-time-difference [h1 m1 h2 m2]
  (let [hours (- h2 h1)
        mins (- m2 m1)
        hours (if (< mins 0) (- hours 1) hours)
        mins (if (< mins 0) (+ 60 mins) mins)
        hours (if (< hours 0) (+ 24 hours) hours)
        hours (if (and (= hours 0) (= mins 0)) 24 hours)]
    [hours mins]))

(defn -main []
  (let [input (clojure.string/split (read-line) #" ")
        [h1 m1 h2 m2] (mapv #(Integer. %) input)
        [hours mins] (calculate-time-difference h1 m1 h2 m2)]
    (println (str "O JOGO DUROU " hours " HORA(S) E " mins " MINUTO(S)"))))

(-main)